rpc-rs/src/provider.rs

```diff
@@ -494,18 +467,18 @@ impl HostBridge {
                 },
             )
             .instrument(tracing::debug_span!("dispatch", public_key = %inv.origin.public_key, operation = %inv.operation))
-            .await;
-
-        #[cfg(feature = "prometheus")]
-        match &rc {
-            Err(_) => {
+            .await {
+            Err(e) => {
+                #[cfg(feature = "prometheus")]
                 self.rpc_client.stats.rpc_recv_err.inc();
-            }
+                Err(e)
+            },
             Ok(vec) => {
+                #[cfg(feature = "prometheus")]
                 self.rpc_client.stats.rpc_recv_resp_bytes.inc_by(vec.len() as u64);
+                Ok(vec)
```

Comment:
Nit, but I think I liked the way this was before. With the prometheus feature disabled, this becomes
```rust
match ... {
  Err(e) => Err(e),
  Ok(vec) => Ok(vec)
}
```

But the reader has to visually parse that. Whereas before, the entire match block can be ignored if the flag isn't enabled

Reply:
I think it's doing what you want- you might have gotten an incomplete snapshot?

 https://github.com/wasmCloud/weld/blob/faf77a724876423815a1b1c1d7aa1730040ba7c8/rpc-rs/src/provider.rs#L476